tools/cellxgene_census_builder/src/cellxgene_census_builder/util.py

```diff
@@ -41,12 +48,41 @@ def urlcat(base: str, *paths: str) -> str:
     return url
 
 
-def env_var_init(args: CensusBuildArgs) -> None:
+def env_var_init() -> None:
     """
     Set environment variables as needed by dependencies, etc.
+
+    This controls thread allocation for worker (child) processes. It is executed too
+    late to influence __init__ time thread pool allocations for the main process.
     """
+
+    # Each of these control thread-pool allocation for commonly used packages that
+    # may be pulled into our environment, and which have import-time pool allocation.
+    # Most do import time thread pool allocation equal to host CPU count, which can
+    # result in excessive unused thread pools on high CPU machines.
+    #
+    # Where we are confident we have no performance dependency related to their concurrency,
+    # set their pool size to "1". Otherwise set to something useful.
+    #
+    # OMP_NUM_THREADS: OpenMp,
+    # OPENBLAS_NUM_THREADS: OpenBLAS,
+    # MKL_NUM_THREADS: Intel MKL,
+    # VECLIB_MAXIMUM_THREADS: Accelerate,
+    # NUMEXPR_NUM_THREADS: NumExpr
+
     if "NUMEXPR_MAX_THREADS" not in os.environ:
-        os.environ["NUMEXPR_MAX_THREADS"] = str(min(1, cpu_count() // 2))
+        # ref: https://numexpr.readthedocs.io/en/latest/user_guide.html#threadpool-configuration
+        # In particular, the docs state that >8 threads is not helpful except in extreme circumstances.
+        os.environ["NUMEXPR_MAX_THREADS"] = str(min(8, max(1, cpu_count() // 2)))
+
+    for env_name in [
+        "OMP_NUM_THREADS",
+        "OPENBLAS_NUM_THREADS",
+        "MKL_NUM_THREADS",
+        "VECLIB_MAXIMUM_THREADS",
+    ]:
+        if env_name not in os.environ:
+            os.environ[env_name] = "1"
```

Comment:
any value in logging the pool size hints put into effect here?

Reply:
I don't see any value.  Can you describe the value you see?

Reply:
If ever performance issues start to crop again, I always find that having logged all config can help troubleshoot, showing that some config is not as expected, etc. Logging config is more of a "good practice", but I don't have a specific need in mind other than that.

Reply:
OK, I'll add something.  Please review as I'm still not sure what info would have real-world utility

Reply:
I don't feel strongly; it was just a thought.  This looks fine, but if you do keep it in, also add for `NUMEXPR_MAX_THREADS`